lib/maths/unittest/CMathsMemoryTest.cc

```diff
@@ -24,14 +25,16 @@ BOOST_AUTO_TEST_SUITE(CMathsMemoryTest)
 using namespace ml;
 using namespace maths;
 
+const ml::maths::CModelAddSamplesParams::TModelChangeCallback NOOP;
```

Comment:
This is a null function object and this test code will throw. But I think it is better to handle defaulting in the function definition.